### DISCUSSION TITLE
Pass VCard, WIFI data to QRCode

### DISCUSSION THREAD

**Discussion by menjilx** [chillerlan/php-qrcode#115] [Q&A]
How to pass the vcard and WIFI data to QRCode?

I tried to pass it as ARRAY but it will only accept STRING.

Thank you

**Comment by codemasher** [comment]
That's right, the encoder only accepts strings, so you will need to prepare the data on your own. There are no plans to add specific protocols as this is not the focus of this library and an entire can of worms altogether.

**Reply by menjilx** [reply]
Thank you so much for the response.

**Comment by menjilx** [comment]
I got the WIFI to QR Code code working. Here's the code below.

`
// WIFI QR Code Test
$ssid = 'MyWifiNetwork';
$type = "WPA";
$hidden = "false";
$pass = "123456";


$wifi = 'WIFI:S:'.$ssid.';'.'T:'.$type.';'.'H:'.$hidden.';'.'P:'.$pass.';'.';;';

$qrOutputInterface = new QRImageWithLogo(
    $options,
    (new QRCode($options))->getMatrix($wifi),
);
`

**Reply by LukmanNulHakim15** [reply]
Hi @menjilx ,
What do you use library for generate qr code ?
any reference for me

**Comment by alamadcs** [comment]
how to input VCARD?

**Reply by codemasher** [reply]
See https://php-qrcode.readthedocs.io/en/main/Appendix/URI-Content.html#contact-information-vcard
